spec/integ/megolm-integ.spec.ts

```diff
@@ -498,60 +488,21 @@ describe("megolm", () => {
         // mark the device as known, and resend.
         aliceTestClient.client.setDeviceKnown("@bob:xyz", "DEVICE_ID");
 
-        let inboundGroupSession: Olm.InboundGroupSession;
-        aliceTestClient.httpBackend
-            .when("PUT", "/sendToDevice/m.room.encrypted/")
-            .respond(200, function (_path, content: any) {
-                const m = content.messages["@bob:xyz"].DEVICE_ID;
-                const ct = m.ciphertext[testSenderKey];
-                const decrypted = JSON.parse(p2pSession.decrypt(ct.type, ct.body));
-
-                expect(decrypted.type).toEqual("m.room_key");
-                inboundGroupSession = new Olm.InboundGroupSession();
-                inboundGroupSession.create(decrypted.content.session_key);
-                return {};
-            });
-
-        aliceTestClient.httpBackend.when("PUT", "/send/").respond(200, (_path, content: IContent) => {
-            const ct = content.ciphertext;
-            const r: any = inboundGroupSession.decrypt(ct);
-            logger.log("Decrypted received megolm message", r);
-
-            expect(r.message_index).toEqual(0);
-            const decrypted = JSON.parse(r.plaintext);
-            expect(decrypted.type).toEqual("m.room.message");
-            expect(decrypted.content.body).toEqual("test");
-
-            return { event_id: "$event_id" };
-        });
-
         const room = aliceTestClient.client.getRoom(ROOM_ID)!;
         const pendingMsg = room.getPendingEvents()[0];
 
         await Promise.all([
             aliceTestClient.client.resendEvent(pendingMsg, room),
-
-            // the crypto stuff can take a while, so give the requests a whole second.
-            aliceTestClient.httpBackend.flushAllExpected({ timeout: 1000 }),
+            expectSendKeyAndMessage(aliceTestClient.httpBackend, "@bob:xyz", testOlmAccount, p2pSession),
```

Comment:
Well, I've written another test that's going to use it, but I don't entirely disagree with you: given it's only two function calls maybe it would be clearer to inline it.

Reply:
done in 05b8434